### ISSUE TITLE
skills: Pipeline Imbalance fires on NVTX phase labels and recommends repartitioning a pipeline that may not exist

### ISSUE THREAD

**Issue by rich7420** [GindaChen/nsys-ai#602]
### Subsystem

skills (`root_cause_matcher` — `_check_pipeline_imbalance`)

### What happened, and what you expected instead

`_check_pipeline_imbalance` compares compute time across whatever `nvtx_layer_breakdown` returns and calls the spread "Pipeline Imbalance". Nothing requires those regions to be pipeline stages. On an ordinary PyTorch capture they are phase labels, and phases are *supposed* to differ:

```
pattern       : Pipeline Imbalance
evidence      : Compute time varies 120.0× across layers.
                Heaviest: 'backward' (240.0ms), lightest: 'data_load' (2.0ms)
recommendation: Rebalance pipeline stage partitioning, or investigate if the
                heavy layer has suboptimal kernel configuration ...
```

That is a single-GPU training step with no pipeline parallelism anywhere. `backward` taking 120× `data_load` is what a healthy iteration looks like, and "rebalance pipeline stage partitioning" is advice about a structure that may not exist.

Expected: the check fires on stages of a pipeline, or says what it is actually comparing.

### How to reproduce

```shell
$ python - <<'PY'
import sys; sys.path.insert(0, "src")
from nsys_ai.skills.builtins.root_cause_matcher import _check_pipeline_imbalance
print(_check_pipeline_imbalance([
    {"nvtx_region": "forward",   "compute_ms": 120.0},
    {"nvtx_region": "backward",  "compute_ms": 240.0},
    {"nvtx_region": "optimizer", "compute_ms": 15.0},
    {"nvtx_region": "data_load", "compute_ms": 2.0},
]))
PY
```

### Why it happens

`layer_data` is whatever `nvtx_layer_breakdown` produced (`root_cause_matcher.py:488`). The function's own docstring says "across NVTX layers", but an NVTX region is not a layer — it is whatever the annotator named, and the two most common annotation styles in the wild are per-phase and per-module.

### Suggested fix

The check needs a reason to believe it is looking at peers. Options, roughly in increasing cost:

1. **Require sibling regions at the same NVTX depth under a common parent.** Phases of an iteration are siblings, but so are pipeline stages — this alone does not separate them, though it does stop the check comparing a range to its own child.
2. **Require repetition.** Pipeline stages recur with similar shape across iterations; a `data_load` vs `backward` comparison does not survive asking "do these look like instances of the same thing?".
3. **Name what it compared, and soften the recommendation** when it cannot tell. "Compute varies 120× across the NVTX regions `forward`, `backward`, `optimizer`, `data_load`" is true and lets the reader judge; "rebalance pipeline stage partitioning" asserts a diagnosis the evidence does not support.

(3) is worth doing regardless of which of the others lands, because the recommendation is the part that misleads.

### Profile shape

Any single-GPU PyTorch capture with phase-style NVTX annotation — which is what `torch.cuda.nvtx.range_push("forward")` produces, and what most tutorials show.
